_common/qiskit/execute.py

```diff
@@ -754,9 +756,9 @@ def execute_circuit(circuit):
                 logger.info(f'Running trans_qc, shots={shots}')
                 st = time.time() 
 
-                if use_sessions:
+                if sampler:
```

Comment:
`use_sessions` is a boolean variable that specifies if one uses the `Session` mode to execute circuits, whereas `sampler` is an instance of the qiskit Primitive to run circuits. Here, the correct logic seems to be checking if a `sampler` primitive is instantiated; otherwise, use the old `backend.run()` method.